File: python/test/CleanAPITests.py
import volar, pprint, ConfigParser

"""
Clears out test broadcasts and playlists created by TestEditBroadcast and TestEditPlaylist
"""

# load settings
c = ConfigParser.ConfigParser()
c.read('sample.cfg')	#note that this file is only for use with this script.  however, you can copy its contents and this code to use in your own scripts
base_url = c.get('settings','base_url')
api_key = c.get('settings','api_key')
secret = c.get('settings','secret')
v = volar.Volar(base_url = base_url, api_key = api_key, secret = secret)

response = v.broadcasts({'site': 'volar', 'title': 'api_test'})
for bcast in response['broadcasts']:
	response = v.broadcast_delete({'site': 'volar', 'id': bcast['id']})
	if (response['success']):
		print('Deleted Broadcast #'+bcast['id'])
	else:
		print('Failed to Delete Broadcast #'+bcast['id'])

response = v.broadcasts({'site': 'volar', 'title': 'api_test_2'})
for bcast in response['broadcasts']:
	response = v.broadcast_delete({'site': 'volar', 'id': bcast['id']})
	if (response['success']):
		print('Deleted Broadcast #'+bcast['id'])
	else:
		print('Failed to Delete Broadcast #'+bcast['id'])

response = v.playlists({'site': 'volar', 'title': 'api_test'})
for plist in response['playlists']:
	response = v.playlist_delete({'site': 'volar', 'id': plist['id']})
	if (response['success']):
		print('Deleted Playlist #'+plist['id'])
	else:
		print('Failed to Delete Playlist #'+plist['id'])

response = v.playlists({'site': 'volar', 'title': 'api_test_2'})
for plist in response['playlists']:
	response = v.playlist_delete({'site': 'volar', 'id': plist['id']})
	if (response['success']):
		print('Deleted Playlist #'+plist['id'])
	else:
		print('Failed to Delete Playlist #'+plist['id'])

File: php/test/CleanAPITests.php
<?php

require('../Volar.php');
require('../test_config.php');
$server = VOLAR_BASE_URL;
$api_key = VOLAR_API_KEY;
$secret_key = VOLAR_SECRET_KEY;

$v = new Volar($api_key, $secret_key, $server);

/*foreach($argv as $argument)
{
$params = array("site" => "volar", "id" => $argument);
$result = $v->broadcast_delete($params);

if($result["success"])
	echo("Broadcast #".$argument." Deleted\n");
else
	echo("Failed to delete broadcast #".$argument."\n");
}*/
$params = array("site" => "volar", "title" => "api_test");
$result = $v->broadcasts($params);

foreach($result["broadcasts"] as $bcast)
{
	$delete_params = array("site" => "volar", "id" => $bcast["id"]);
	$result = $v->broadcast_delete($delete_params);
	if($result["success"])
		echo("Broadcast #".$bcast["id"]." Deleted\n");
	else
		echo("Failed to delete broadcast #".$bcast["id"]."\n");

}

$params = array("site" => "volar", "title" => "api_test_2");
$result = $v->broadcasts($params);

foreach($result["broadcasts"] as $bcast)
{
	$delete_params = array("site" => "volar", "id" => $bcast["id"]);
	$result = $v->broadcast_delete($delete_params);
	if($result["success"])
		echo("Broadcast #".$bcast["id"]." Deleted\n");
	else
		echo("Failed to delete broadcast #".$bcast["id"]."\n");

}

$params = array("site" => "volar", "title" => "api_test");
$result = $v->playlists($params);

foreach($result["playlists"] as $bcast)
{
	$delete_params = array("site" => "volar", "id" => $bcast["id"]);
	$result = $v->playlist_delete($delete_params);
	if($result["success"])
		echo("playlist #".$bcast["id"]." Deleted\n");
	else
		echo("Failed to delete playlist #".$bcast["id"]."\n");

}

$params = array("site" => "volar", "title" => "api_test_2");
$result = $v->playlists($params);

foreach($result["playlists"] as $bcast)
{
	$delete_params = array("site" => "volar", "id" => $bcast["id"]);
	$result = $v->playlist_delete($delete_params);
	if($result["success"])
		echo("Playlist #".$bcast["id"]." Deleted\n");
	else
		echo("Failed to delete playlist #".$bcast["id"]."\n");

}
?>
